lib/new_sets.bzl

```diff
@@ -0,0 +1,197 @@
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Skylib module containing common hash-set algorithms."""
+
+load(":dicts.bzl", "dicts")
+
+
+def _set(elements=None):
+  """Creates a new set.
+
+  All elements must be hashable.
+
+  Args:
+    elements: Optional enum to construct the set out of.
+
+  Returns:
+    A set containing the passed in values.
+  """
+  elements = elements if elements else []
+  return struct(_values = {e: None for e in elements})
+
+
+def _copy(s):
+  """Creates a new set from another set.
+
+  Args:
+    s: A set.
```

Comment:
So, continuing the line of discussion above, I might write this throughout as "A set, as returned by `sets.make`."